app/mci/pkg/gcp/loadbalancer/loadbalancersyncer.go

```diff
@@ -46,32 +49,115 @@ const (
 	// Prefix used by the namer to generate names.
 	// This is used to identify resources created by this code.
 	mciPrefix = "mci1"
+
+	// instanceGroupsAnnotationKey is the annotation key used by gclb ingress
+	// controller to specify the name and zone of instance groups created
+	// for the ingress.
+	// TODO(nikhiljindal): Refactor the code in kubernetes/ingress-gce to
+	// export it and then reuse it here.
+	instanceGroupsAnnotationKey = "ingress.gcp.kubernetes.io/instance-groups"
 )
 
 type LoadBalancerSyncer struct {
 	lbName string
 	hcs    healthcheck.HealthCheckSyncerInterface
+	bss    backendservice.BackendServiceSyncerInterface
 	client kubeclient.Interface
+	igp    ingressig.InstanceGroups
 }
 
 func NewLoadBalancerSyncer(lbName string, client kubeclient.Interface, cloud *gce.GCECloud) *LoadBalancerSyncer {
 	namer := utilsnamer.NewNamer(mciPrefix, lbName)
 	return &LoadBalancerSyncer{
 		lbName: lbName,
 		hcs:    healthcheck.NewHealthCheckSyncer(namer, cloud),
+		bss:    backendservice.NewBackendServiceSyncer(namer, cloud),
 		client: client,
+		igp:    cloud,
 	}
 }
 
 func (l *LoadBalancerSyncer) CreateLoadBalancer(ing *v1beta1.Ingress) error {
+	var err error
 	ports := l.ingToNodePorts(ing)
-	// Create health check to be used by the backend service.
-	if err := l.hcs.EnsureHealthCheck(l.lbName, ports); err != nil {
+	// Create health checks to be used by the backend service.
+	healthChecks, hcErr := l.hcs.EnsureHealthCheck(l.lbName, ports)
+	if hcErr != nil {
+		// Keep aggregating errors and return all at the end, rather than giving up on the first error.
+		err = multierror.Append(err, hcErr)
+	}
+	igs, namedPorts, err := l.getIGsAndNamedPorts(ing)
```

Comment:
you use multierror above, but then you always overwrite 'err' here and in the next if block...

Reply:
Fixed